pkg/network/network.go

```diff
@@ -0,0 +1,146 @@
+package network
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/skupperproject/skupper/api/types"
+	"strconv"
+	"strings"
+)
+
+type SkupperStatus struct {
+	VanStatus *types.VanStatusInfo
+}
+
+func (s *SkupperStatus) GetServiceSitesMap() map[string][]types.SiteStatusInfo {
+
+	mapServiceSites := make(map[string][]types.SiteStatusInfo)
+
+	for _, site := range s.VanStatus.SiteStatus {
+
+		for _, listener := range site.RouterStatus[0].Listeners {
+			if mapServiceSites[listener.Name] != nil {
+				serviceSites := mapServiceSites[listener.Name]
+
+				serviceSites = append(serviceSites, site)
+				mapServiceSites[listener.Name] = serviceSites
+			} else {
+				mapServiceSites[listener.Name] = []types.SiteStatusInfo{site}
+			}
+		}
+	}
+
+	return mapServiceSites
+}
+
+func (s *SkupperStatus) GetServiceLabelsMap(services []*types.ServiceInterface) map[string]map[string]string {
+
+	mapServiceLabels := make(map[string]map[string]string)
+
+	for _, svc := range services {
+		if svc.Labels != nil {
+			for _, port := range svc.Ports {
+				serviceName := svc.Address + ":" + strconv.Itoa(port)
+				mapServiceLabels[serviceName] = svc.Labels
+			}
+		}
+
+	}
+
+	return mapServiceLabels
+}
+
+func (s *SkupperStatus) GetSiteTargetsMap() map[string]map[string]types.ConnectorInfo {
+
+	mapSiteTargets := make(map[string]map[string]types.ConnectorInfo)
+
+	for _, site := range s.VanStatus.SiteStatus {
+
+		for _, connector := range site.RouterStatus[0].Connectors {
```

Comment:
Same here...

Reply:
Solved in 3371b9e46ea653a9f283ff1dd6c3bcff3e0ef4ae